compiler/rustc_codegen_llvm/src/common.rs

```diff
@@ -108,7 +108,11 @@ impl<'ll> CodegenCx<'ll, '_> {
 
     pub fn const_get_elt(&self, v: &'ll Value, idx: u64) -> &'ll Value {
         unsafe {
-            assert_eq!(idx as c_uint as u64, idx);
+            assert!(
+                idx <= c_uint::MAX as u64,
+                "LLVMGetAggregateElement doesn't support indices greater than {}",
+                c_uint::MAX
+            );
             let r = llvm::LLVMGetAggregateElement(v, idx as c_uint).unwrap();
```

Comment:
Could you use `c_uint::try_from` here? Seems cleaner than an `as` cast

Reply:
sure